feat: Advanced authorization (#2668)
codenvy/che-docs: modules/administration-guide/pages/configuring-advanced-authorization.adoc

:_content-type: PROCEDURE
:description: Configuring advanced authorization
:keywords: authorization, user, group
:navtitle: Configuring advanced authorization
// :page-aliases:

[id="configuring-advanced-authorization"]
= Configuring advanced authorization

You can determine which users and groups are allowed to access {prod-short}.

.Prerequisites

* An active `{orch-cli}` session with administrative permissions to the destination {orch-name} cluster. See {orch-cli-link}.

.Procedure

. Configure the `CheCluster` Custom Resource. See xref:using-the-cli-to-configure-the-checluster-custom-resource.adoc[].
+
[source,yaml,subs="+quotes,+attributes"]
----
spec:
  networking:
    auth:
      advancedAuthorization:
        allowUsers:
          - __<allow_users>__ <1>
        allowGroups:
          - __<allow_groups>__ <2>
        denyUsers:
          - __<deny_users>__ <3>
        denyGroups:
          - __<deny_groups>__ <4>
----
<1> List of users allowed to access {prod}.
<2> List of groups of users allowed to access {prod} (for {ocp} only).
<3> List of users denied access to {prod}.
<4> List of groups of users denied to access {prod} (for {ocp} only).
+

> To allow a user to access {prod-short}, add him to the `allowUsers` list or a group where he is member of to `allowGroups` list.

Reviewer: ```suggestion
To allow a user to access {prod-short}, add them to the `allowUsers` list. Alternatively, choose a group the user is a member of and add the group to the `allowGroups` list.
```

> To deny a user to access {prod-short}, add him to the `denyUsers` list or a group where he is member of to `denyGroups` list.

Reviewer: ```suggestion
To deny a user access to {prod-short}, add them to the `denyUsers` list. Alternatively, choose a group the user is a member of and add the group to the `denyGroups` list.
```

> If user is in both `allow` and `deny` lists, then he is denied access to {prod-short}.

Reviewer: ```suggestion
If the user is on both `allow` and `deny` lists, they are denied access to {prod-short}.
```

+

> If `allowUsers` and `allowGroups` are empty, then all users are allowed to access {prod-short} except ones from `deny` lists.

Reviewer: ```suggestion
If `allowUsers` and `allowGroups` are empty, all users are allowed to access {prod-short} except the ones on the `deny` lists.
```

> if `denyUsers` and `denyGroups` are empty, then only users from `allow` lists are allowed to access {prod-short}.

Reviewer: ```suggestion
If `denyUsers` and `denyGroups` are empty, only the users from `allow` lists are allowed to access {prod-short}.
```

+

> If `allow` and `deny` lists are empty, then all users are allowed to access {prod-short}.

Reviewer: ```suggestion
If both `allow` and `deny` lists are empty, all users are allowed to access {prod-short}.
```

:_content-type: PROCEDURE
:description: Configuring advanced authorization
:keywords: authorization, user, group
:navtitle: Configuring advanced authorization
// :page-aliases:

[id="configuring-advanced-authorization"]
= Configuring advanced authorization

> You can determine which users and groups are allowed to access {prod-short}.

Reviewer 2: @tolusha Should we clarify that this is an OpenShift feature, or this would work on vanilla k8s as well?

.Prerequisites

* An active `{orch-cli}` session with administrative permissions to the destination {orch-name} cluster. See {orch-cli-link}.

.Procedure

. Configure the `CheCluster` Custom Resource. See xref:using-the-cli-to-configure-the-checluster-custom-resource.adoc[].
+
[source,yaml,subs="+quotes,+attributes"]
----
spec:
  networking:
    auth:
      advancedAuthorization:
        allowUsers:
          - __<allow_users>__ <1>
        allowGroups:
          - __<allow_groups>__ <2>
        denyUsers:
          - __<deny_users>__ <3>
        denyGroups:
          - __<deny_groups>__ <4>
----
<1> List of users allowed to access {prod}.
<2> List of groups of users allowed to access {prod} (for {ocp} only).
<3> List of users denied access to {prod}.
<4> List of groups of users denied to access {prod} (for {ocp} only).

> +

Reviewer 2: ok, looks liks on k8s one can only use `allowUsers` / `denyUsers`, whereas `allowGroups` / `denyGroups` is OpenShift specific.
